src/DonorProfiles/resources/js/app/utils/index.js

```diff
@@ -2,5 +2,47 @@ import { addTab } from '../store/actions';
 
 export const registerTab = ( tab ) => {
 	const { dispatch } = window.giveDonorProfile.store;
-	dispatch( addTab( tab ) );
+
+	// Validate the tab object
+	if ( isValidTab( tab ) === true ) {
+		dispatch( addTab( tab ) );
+	} else {
+		return null;
+	}
+};
+
+const isValidTab = ( tab ) => {
+	const tabPropTypes = {
+		slug: 'string',
+		icon: 'string',
+		label: 'string',
+		content: 'function',
+	};
+
+	const isValid = Object.keys( tabPropTypes ).reduce( ( acc, key ) => {
+		if ( typeof tab[ key ] !== tabPropTypes[ key ] ) {
+			/* eslint-disable no-console */
```

Comment:
You should be able to use `eslint-disable-next-line` instead of wrapping `console` in an enable/disable.

Reply:
Ah yes! Cleaner, ref: https://github.com/impress-org/givewp/pull/5455/commits/edf9d0c3ecdedc213bf62a2ed0446d1eda73c13e